lib/src/lints/avoid_using_api/avoid_using_api_linter.dart

```diff
@@ -280,4 +288,96 @@ class AvoidUsingApiLinter {
       reporter.atNode(node.identifier, entryCode);
     });
   }
+
+  void _banDefaultConstructor(
+    String className,
+    String source,
+    LintCode entryCode,
+  ) {
+    context.registry.addInstanceCreationExpression((node) {
+      final constructorName = node.constructorName.type.name2.lexeme;
+      if (constructorName != className || node.constructorName.name != null) {
+        return;
+      }
+
+      final sourcePath =
+          node.constructorName.type.element2?.library2?.uri.toString();
+      if (sourcePath == null || !_matchesSource(sourcePath, source)) {
+        return;
+      }
+
+      reporter.atNode(node, entryCode);
+    });
+  }
+
+  /// Lints usages of a named parameter from a given source
+  void banUsageWithSpecificNamedParameter(
+    LintCode entryCode,
+    String identifier,
+    String namedParameter,
+    String className,
+    String source,
+  ) {
+    context.registry.addMethodInvocation((node) {
+      final methodName = node.methodName.name;
+      if (methodName != identifier) return;
+
+      final enclosingElement = node.methodName.element?.enclosingElement2;
+      if (enclosingElement == null || enclosingElement.name3 != className) {
+        return;
+      }
+
+      if (!_hasNamedParameter(node.argumentList, namedParameter)) {
+        return;
+      }
+
+      final libSource = enclosingElement.library2;
+      if (libSource == null) {
+        return;
+      }
+
+      final sourcePath = libSource.uri.toString();
+      if (!_matchesSource(sourcePath, source)) {
+        return;
+      }
+
+      reporter.atNode(node.methodName, entryCode);
+    });
+
+    context.registry.addInstanceCreationExpression((node) {
+      final String? expectedConstructorName;
+      if (identifier == defaultConstructorIdentifier) {
+        expectedConstructorName = null;
+      } else {
+        expectedConstructorName = identifier;
+      }
+
+      final actualClassName = node.constructorName.type.name2.lexeme;
+      if (actualClassName != className) {
+        return;
+      }
+
+      if (node.constructorName.name?.name != expectedConstructorName) {
+        return;
+      }
+
+      if (!_hasNamedParameter(node.argumentList, namedParameter)) {
+        return;
+      }
+
+      final sourcePath =
+          node.constructorName.type.element2?.library2?.uri.toString();
+      if (sourcePath == null || !_matchesSource(sourcePath, source)) {
+        return;
+      }
+
+      reporter.atNode(node, entryCode);
+    });
+  }
+
+  bool _hasNamedParameter(ArgumentList argumentList, String namedParameter) =>
```

Comment:
```suggestion
  bool _containsNamedParameter(ArgumentList argumentList, String namedParameter) =>
```
Would it be a little bit clearer?